charts/federation-v2/charts/controllermanager/templates/deployment.yaml

```diff
@@ -22,20 +22,8 @@ spec:
     spec:
       containers:
       - args:
-{{- if .Values.featureGates }}
-        - --feature-gates={{ .Values.featureGates }}
-{{- end }}
 {{- if .Values.global.limitedScope }}
         - --limited-scope={{ .Values.global.limitedScope }}
-        - --federation-namespace=$(FEDERATION_NAMESPACE)
```

Comment:
Please restore this line and use it to configure the controller manager without condition.  The `--federation-namespace` arg should be retained to allow this.  

My understanding is that it is preferable to communicate values from the downward API via an arg rather than an environment variable.  Initial implementations of the controller manager used environment variables until @pmorie taught me the errors of my ways.

Reply:
@marun, 
What is the issue for using environment variable for namespace? I think in helm chart, it is fine to use environment variable instead of command line options. Because the dependency (environment variable and federation config) should be always there. And it is way to use downward API.

Reply:
It's not the helm chart we're talking about - it's the `controller-manager` binary.  Environment variables are not as easily discoverable as explicit command-line arguments and easy to screw up (since they can be set separate from command invocation).  Unless we have a specific reason to use environment variables - and we arguably do not - we shouldn't use them.

Reply:
I'm ok to add it back because both way works.
 
My question is : what is the reason that downward api with using environment variable is arguable. 
It is a proposed way 
> to expose Pod and Container fields to a running Container.

Reply:
Had a sync-up with marun: environment variable is not preferred as controller configuration from command line. `mybinary --my-var=${MY_VAR}` is better than `MY_VAR=foo mybinary`